lib/project-overlays.nix

```diff
@@ -7,16 +7,16 @@
   devshell = final: prev: {
     devshell = let
     in {
-      packages = final.shell.nativeBuildInputs
-      # Cannot add the whole final.shell.buildInputs list because many collide with each other when fused.
-      # So we only add what is really used (by pkg-config):
-      ++ map (p: p.dev or p) (lib.concatLists (lib.concatMap (p: p.components.library.pkgconfig or [] ++ p.components.setup.pkgconfig or [] ++ lib.concatMap (c: lib.concatMap (a: a.pkgconfig) (lib.attrValues c)) (lib.attrValues (removeAttrs p.components ["library" "setup"])))
-        (lib.attrValues final.pkg-set.config.packages)))
+      packages = (lib.filter lib.isDerivation (final.shell.nativeBuildInputs
       # devshell does not use pkgs.mkShell / pkgs.stdenv.mkDerivation,
       # so we need to explicit required dependencies which
       # are provided implicitely by stdenv when using the normal shell:
-      ++ (lib.filter lib.isDerivation final.shell.stdenv.defaultNativeBuildInputs)
-      ++ lib.optional final.shell.stdenv.targetPlatform.isGnu final.pkgs.buildPackages.binutils;
+      ++ final.shell.stdenv.defaultNativeBuildInputs))
+      ++ lib.optional final.shell.stdenv.targetPlatform.isGnu final.pkgs.buildPackages.binutils
+      # Cannot add the whole final.shell.buildInputs list because many collide with each other when fused.
```

Comment:
Can you elaborate on this a bit? It seems like it would simplify things a lot if we could just pass through the full build inputs list?

Reply:
devshell fuse all provided packages into a single derivation. But since many entries in buildInputs have some files in common, this fails.
I'm testing an alternative that avoid this issue entirely though: https://github.com/input-output-hk/haskell.nix/commit/ea40f5cba8460f66250b5f38a19140e25188ae6b
I'll open a pr on it if it works correctly.

Reply:
Is this something we can ask for help with upstream? Is there a `devshell` issue about this?

Reply:
there is this bug that point to that limitation: https://github.com/numtide/devshell/issues/163 
And I think `lib.lowPrio` and `lib.hiPrio` won't solve this this this would exclude whole package not just the duplicate files.